src/Task.gren

```diff
@@ -125,6 +125,32 @@ map func taskA =
     taskA
         |> andThen (\a -> succeed (func a))
 
+{-| Apply the results of two tasks together if they succeed. Both tasks are run concurrently, and if either task fails the whole task fails:
+
+    succeed ((+) 2)
+        |> andMap (succeed 3)
+    --> succeed 5
+
+    fail "oh dear"
+        |> andMap (succeed 3)
```

Comment:
Since `fail` and `succeed` can be different types, this is valid, but the type mismatch threw me off at first glance.

Reply:
Do you think it would be clearer like?:

```elm
succeed ((+) 2)
    |> andMap (succeed 3)
--> succeed 5

succeed ((+) 2)
    |> andMap (fail "oh dear")
--> fail "oh dear"
```